Data/ByteString/Base64/URL.hs

```diff
@@ -31,13 +32,17 @@ import Foreign.ForeignPtr (ForeignPtr)
 -- | Encode a string into base64url form.  The result will always be a
 -- multiple of 4 bytes in length.
 encode :: ByteString -> ByteString
-encode = encodeWith (mkEncodeTable alphabet)
+encode = encodeWith True (mkEncodeTable alphabet)
 
--- | Decode a base64url-encoded string.  This function strictly follows
--- the specification in
--- <http://tools.ietf.org/rfc/rfc4648 RFC 4648>.
+-- | Encode a string into unpadded base64url form.
+encodeUnpadded :: ByteString -> ByteString
+encodeUnpadded = encodeWith False (mkEncodeTable alphabet)
+
+-- | Decode a base64url-encoded string applying padding if necessary.
+-- This function follows the specification in <http://tools.ietf.org/rfc/rfc4648 RFC 4648>
+-- and in <https://tools.ietf.org/html/rfc7049#section-2.4.4.2 RFC 7049 2.4>
 decode :: ByteString -> Either String ByteString
-decode = decodeWithTable decodeFP
+decode = decodeWithTable True decodeFP
```

Comment:
In contrast to `Base64.hs`, Base64url is optionally padded, and will be treated as such.